streamlit/mmm-explainer/Visualise_Priors.py

```diff
@@ -0,0 +1,252 @@
+    fig.update_layout(height=500, width=1000)
+    st.plotly_chart(fig, use_container_width=True)
+
+# -------------------------- GAMMA DISTRIBUTION -------------------------
+with tab5:
+    st.header(":red[Gamma Distribution]")
+
+    st.subheader(":red[User Inputs]")
+    alpha = st.number_input("Alpha for Gamma", value=1.0, min_value=0.01)
+    beta = st.number_input("Beta for Gamma (rate)", value=1.0, min_value=0.01)
+
+    # Draw samples
+    samples = pf.draw_samples_from_prior(pm.Gamma, alpha=alpha, beta=beta, seed=seed)
+
+    # Plot distribution
+    fig = pf.plot_prior_distribution(samples, title="Gamma Distribution Samples")
+    fig.update_layout(height=500, width=1000)
+    st.plotly_chart(fig, use_container_width=True)
+
+# -------------------------- POISSON DISTRIBUTION -------------------------
+with tab6:
+    st.header(":grey[Poisson Distribution]")
+
+    st.subheader(":grey[User Inputs]")
+    mu = st.number_input("Lambda (mu)", value=1.0, min_value=0.01)
+
+    # Draw samples
+    samples = pf.draw_samples_from_prior(pm.Poisson, mu=mu, seed=seed)
+
+    # Plot distribution
+    fig = pf.plot_prior_distribution(samples, title="Poisson Distribution Samples")
+    fig.update_layout(height=500, width=1000)
+    st.plotly_chart(fig, use_container_width=True)
+
+# -------------------------- BERNOULLI DISTRIBUTION -------------------------
+with tab7:
+    st.header(":rainbow[Bernoulli Distribution]")
+
+    st.subheader(":rainbow[User Inputs]")
+    p = st.number_input(
+        "Probability of Success (p)", value=0.5, min_value=0.0, max_value=1.0
+    )
+
+    # Draw samples
+    samples = pf.draw_samples_from_prior(pm.Bernoulli, p=p, seed=seed)
+
+    # Plot distribution
+    fig = pf.plot_prior_distribution(samples, title="Bernoulli Distribution Samples")
+    fig.update_layout(height=500, width=1000)
+    st.plotly_chart(fig, use_container_width=True)
+
+# -------------------------- EXPONENTIAL DISTRIBUTION -------------------------
+with tab8:
+    st.header(":blue[Exponential Distribution]")
+
+    st.subheader(":blue[User Inputs]")
+    lam = st.number_input("Rate (lambda)", value=1.0, min_value=0.01)
+
+    # Draw samples
+    samples = pf.draw_samples_from_prior(pm.Exponential, lam=lam, seed=seed)
+
+    # Plot distribution
+    fig = pf.plot_prior_distribution(samples, title="Exponential Distribution Samples")
+    fig.update_layout(height=500, width=1000)
+    st.plotly_chart(fig, use_container_width=True)
+
+# -------------------------- WEIBULL DISTRIBUTION ---------------------------
+with tab9:
+    st.header(":violet[Weibull Distribution]")
+
+    st.subheader(":violet[User Inputs]")
+    alpha = st.number_input("Shape (alpha)", value=1.5, min_value=0.01)
+    beta = st.number_input("Scale (beta)", value=1.0, min_value=0.01)
+
+    samples = pf.draw_samples_from_prior(pm.Weibull, alpha=alpha, beta=beta, seed=seed)
+    fig = pf.plot_prior_distribution(samples, title="Weibull Distribution Samples")
```

Comment:
This fig and after do not have set width and height. Is there a reason for that? Shouldnt they all be the same?
Also, pulling these out as setting variables at the top would be easier to track

Reply:
Done and done! 😁